diff --git a/node_modules/mysql2/lib/connection_config.js b/node_modules/mysql2/lib/connection_config.js
index ec52051..52ddc36 100644
--- a/node_modules/mysql2/lib/connection_config.js
+++ b/node_modules/mysql2/lib/connection_config.js
@@ -260,11 +260,11 @@ class ConnectionConfig {
   static parseUrl(url) {
     const parsedUrl = new URL(url);
     const options = {
-      host: parsedUrl.hostname,
+      host: decodeURIComponent(parsedUrl.hostname),
       port: parseInt(parsedUrl.port, 10),
-      database: parsedUrl.pathname.slice(1),
-      user: parsedUrl.username,
-      password: parsedUrl.password
+      database: decodeURIComponent(parsedUrl.pathname.slice(1)),
+      user: decodeURIComponent(parsedUrl.username),
+      password: decodeURIComponent(parsedUrl.password)
     };
     parsedUrl.searchParams.forEach((value, key) => {
       try {
